packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol

```diff
@@ -1173,46 +992,57 @@ contract VotingV2 is Staker, OracleInterface, OracleAncillaryInterface, OracleGo
+
             rounds[roundId].gat = gat;
 
             // Store the cumulativeStake at this roundId to work out slashing and voting trackers.
             rounds[roundId].cumulativeStakeAtRound = cumulativeStake;
         }
     }
 
-    // Returns if a given price request, with known votingInstance and currentRoundId is resolved.
-    function _priceRequestResolved(
-        PriceRequest storage priceRequest,
-        VoteInstance storage voteInstance,
-        uint256 currentRoundId
-    ) private returns (bool) {
-        // We are currently either in the voting round for the request or voting is yet to begin.
-        if (currentRoundId <= priceRequest.lastVotingRound) return false;
-
-        // If the request has been previously resolved, return true.
-        if (priceRequest.pendingRequestIndex == UINT64_MAX) return true;
-
-        // Else, check if the price can be resolved.
-        (bool isResolvable, int256 resolvedPrice) =
-            voteInstance.resultComputation.getResolvedPrice(_computeGat(priceRequest.lastVotingRound));
-
-        // If it's not resolvable return false.
-        if (!isResolvable) return false;
-
-        // Else, the request is resolvable. Remove the element from the pending requests and update pendingRequestIndex
-        // within the price request struct to make the next entry into this method a no-op for this request.
-        _removeRequestFromPendingPriceRequests(priceRequest.pendingRequestIndex);
-
-        priceRequest.pendingRequestIndex = UINT64_MAX;
-        emit PriceResolved(
-            priceRequest.lastVotingRound,
-            priceRequest.priceRequestIndex,
-            priceRequest.identifier,
-            priceRequest.time,
-            priceRequest.ancillaryData,
-            resolvedPrice
-        );
-        return true;
+    function _resolveResolvablePriceRequests() private {
+        uint32 currentRoundId = uint32(getCurrentRoundId());
+
+        uint256 index = 0;
+        while (index < pendingPriceRequestsIds.length) {
+            PriceRequest storage request = priceRequests[pendingPriceRequestsIds[index]];
+            if (request.lastVotingRound >= currentRoundId) {
+                ++index;
+                continue;
+            }
+            VoteInstance storage voteInstance = request.voteInstances[request.lastVotingRound];
+            (bool isResolvable, int256 resolvedPrice) =
+                voteInstance.resultComputation.getResolvedPrice(_computeGat(request.lastVotingRound));
+            if (!isResolvable) {
+                if (request.lastVotingRound < currentRoundId) {
+                    request.rollCount += currentRoundId - request.lastVotingRound;
+                    request.lastVotingRound = currentRoundId;
+                    emit PriceRequestRolled(request.identifier, request.time, request.ancillaryData, request.rollCount);
+                    if (request.rollCount == deleteAfterRollCount && !request.isGovernance) {
```

Comment:
```suggestion
                    if (request.rollCount >= deleteAfterRollCount && !request.isGovernance) {
```

Now `rollCount` could be incrementing more than by 1 if nobody had touched the contract for couple of rounds.

Reply:
+1 -- this also covers the case where the deleteAfterRollCount variable is decreased between runs of this method.

Reply:
done.